### PR TITLE
Linux_Ubsan_Chrome builds working

### PR DESCRIPTION
**Pull request by everestmz**
cc: @google/clusterfuzz 

This PR enables reproduction for linux_Ubsan_Chrome builds that don't use gestures, and throws a JobNotSupportedError if they do use gestures. As of now including gestures requires further work.

### CODE REVIEW

File: clusterfuzz/binary_providers.py [hunk 1]
```diff
@@ -278,14 +278,19 @@ def pre_build_steps(self):
 class ChromiumBuilder(GenericBuilder):
   """Builds a specific target from inside a Chromium source repository."""
 
-  def __init__(self, testcase, binary_definition, current, goma_dir):
+  def __init__(self, testcase, binary_definition, current, goma_dir,
+               target=None):
```
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: The target seems redundant here. Below we use target name.

File: clusterfuzz/commands/reproduce.py [hunk 2]
```diff
@@ -61,7 +61,11 @@
             sanitizer='MSAN'),
         'libfuzzer_chrome_ubsan': common.BinaryDefinition(
             binary_providers.ChromiumBuilder, 'CHROMIUM_SRC',
-            sanitizer='UBSAN')}}
+            sanitizer='UBSAN'),
+        'linux_ubsan_chrome': common.BinaryDefinition(
+            binary_providers.ChromiumBuilder, 'CHROMIUM_SRC', 'chrome',
+            sanitizer='UBSAN', target='chromium_builder_asan',
+            extra_runtime_args='--user-data-dir=/tmp/user-profile-data')}}
```
Comment (tanin47) [inline clusterfuzz/commands/reproduce.py]: This is special folder, and we need to nuke the folder every time. I don't mind doing in the testcase on every run.

This seems like we need the third class: Reproducer. (We have Testcase which represents a testcase. We have Builder which represents how we build.)

The reproduce_crash function is basically Reproducer class.

Comment (everestmz) [inline clusterfuzz/commands/reproduce.py]: Ah, I didn't know that. Reproducer sounds like a good idea then.

**Comment by everestmz** [comment]
Thank you, feedback addressed!

**Comment by everestmz** [comment]
also fixes #35

File: clusterfuzz/reproducers.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                   environment=self.environment, exit_on_error=False)
+
+
+class LinuxUbsanChromeReproducer(BaseReproducer):
+  """Adds and extre pre-build step to BaseReproducer."""
+
+  def pre_build_steps(self):
+    """Steps to run before building."""
+
+    user_profile_dir = '/tmp/user-profile-data'
```
Comment (tanin47) [inline clusterfuzz/reproducers.py]: Let's throw in the word 'clusterfuzz' into the folder name.

File: clusterfuzz/reproducers.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    if os.path.exists(user_profile_dir):
+      shutil.rmtree(user_profile_dir)
+    self.args += ' --user-data-dir=/tmp/user-profile-data'
```
Comment (tanin47) [inline clusterfuzz/reproducers.py]: Use the variable user_profile_dir instead of string.

File: clusterfuzz/binary_providers.py [hunk 5]
```diff
@@ -280,12 +280,16 @@ class ChromiumBuilder(GenericBuilder):
 
   def __init__(self, testcase, binary_definition, current, goma_dir):
 
+    target_name = None
     binary_name = binary_definition.binary_name
+    if hasattr(binary_definition, 'target'):
```
Comment (tanin47) [inline clusterfuzz/binary_providers.py]: I start to regret the **kwargs design decision.

I think we should explicitly define `target` and the field we need in BinaryDefinition.

Comment (everestmz) [inline clusterfuzz/binary_providers.py]: I agree, it makes more sense to be explicit.

**Pull request merged**